activerecord/lib/active_record/autosave_association.rb

```diff
@@ -272,6 +273,10 @@ def changed_for_autosave?
       new_record? || has_changes_to_save? || marked_for_destruction? || nested_records_changed_for_autosave?
     end
 
+    def already_trying_to_save?
+      @saving ||= false
```

Comment:
It's not going to lead to a problem but I find memoization here being unnecessary as it will still assign `false` to @savings` if it's nil or has already been assigned to `false`

I think we can just rely on the variable value being truthy/falsey
```suggestion
      @saving
```

Reply:
yep, I've done it in this way because of CI warnings

moved instantiation to `init_internals` method